### PR TITLE
#481 create a draft_button component

### PR DESCRIPTION
**Pull request by sw00**
Created a draft_button component that listens to `draft:save` and `draft:saved` events to know when to enable and disable itself. This should prevent spamming of the "Save Draft" button resulting in duplicate draft messages being save.

However, there may be other scenarios I've missed on the page.

### CODE REVIEW

File: web-ui/test/spec/mail_view/ui/draft_button.spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+describeComponent('mail_view/ui/draft_button', function(){
+  'use strict';
+
+  describe('draft save button', function(){
+    beforeEach(function(){
+      this.setupComponent('<button></button>');
+    });
+
+    describe('after initialize', function(){
+      it('should be enabled', function(){
```
Comment (giovaneliberato) [inline web-ui/test/spec/mail_view/ui/draft_button.spec.js]: @sw00 just to make sure, the test name should be "should be disabled" right?

Comment (sw00) [inline web-ui/test/spec/mail_view/ui/draft_button.spec.js]: Fixed.

**Comment by jeffhsta** [comment]
Hey @sw00 I saw on Snap-CI that the PR fail with the below error:

``
PhantomJS 1.9.8 (Linux) mixins/with_mail_edit_base when the user is typing in subject or body does not save if mail is sent before the save draft interval number of seconds FAILED
	<div class="value-mismatch">Expected event draft:save to have been triggered on[object HTMLDocument],<div class="value-mismatch">Expected event draft:save not to have been triggered on[object HTMLDocument]
	Error: <div class="value-mismatch">Expected event draft:save to have been triggered on[object HTMLDocument],<div class="value-mismatch">Expected event draft:save not to have been triggered on[object HTMLDocument]
	    at /var/snap-ci/repo/web-ui/test/spec/mixins/with_mail_edit_base.spec.js:49
``

You can go "service" directory and run the command "./go test" to run the python tests and alse go to "web-ui" directory and run the same command to run the JS tests.

Any question, feel free to ask us here or on IRC

**Comment by sw00** [comment]
Can't reproduce it locally which suggests I've introduced some flaky-ness to that feature. I will investigate.

**Comment by fbernitt** [comment]
Looks good to me. I rebased it to the latest HEAD and seems to work as intended.

**Comment by fbernitt** [comment]
Had to rebase before merge and it seems GitHub did not recognise. But this got merged with commit 09d989a6ed1d3fb1641ec63e5c92e4f212882406

**Pull request closed without merging**